Apsitvarkom.DataAccess/IEntityDTORepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Apsitvarkom.DataAccess;
+
+/// <summary>
+/// Interface for <see cref="T" /> class data handling.
+/// </summary>
+public interface IEntityDTORepository<T> where T: class
+{
+    /// <summary>Gets all <see cref="T" /> records from the data source.</summary>
+    /// <returns><see cref="Enumerable" /> of <see cref="T" /> instances.</returns>
+    public Task<IEnumerable<T>> GetAllAsync();
+
+    /// <summary>Gets a single <see cref="T" /> record from the data source by id.</summary>
+    /// <param name="id">Identifier of the requested record.</param>
+    /// <returns>Null if the instance was not found. Otherwise a <see cref="T" /> instance with matching identifier.</returns>
+    public Task<T?> GetByIdAsync(string id);
```

Comment:
What if type `T` does not have an `Id` property, or that property's type is not `string`? You have to keep these things in mind when creating a *generic* interface (you either limit the generic type so that you will be certain that the generic type will have the fields you need or you generalize the methods more).

Maybe we could go for `public Task<T?> GetByPropertyAsync(Predicate<T> propertyCondition);`

That could be used something like this:
```csharp
const string idToSearch = "this-is-an-id";
var dto = _repository.GetByPropertyAsync(x => x.Id == myId);

const int amountToSearch = 42;
var otherDto = _otherTypeRepository.GetByPropertyAsync(x => x.Amount == amountToSearch);
```

Reply:
yuup, makes sense. Just I believe I had to use not predicate, but func there.

Reply:
I would prefer a `Predicate<T>` since it's achieves the same thing as `Func<T, bool>` (represents a function that always returns `bool`), but a quick google search [leads to this](https://stackoverflow.com/a/665513), so we can keep it just for the sake of LINQ's consistency